### PR TITLE
Update dependencies, build fixes

### PR DESCRIPTION
**Pull request by dgladkov**
* Old SDL2 patch removed
* All dependencies updated to the latest version
* Added `build/` to `.gitignore`
* Bumped default OSX SDK to `macosx10.13`
* Added full sysroot path lookup from SDK name string as using just the name (e.g. `macosx10.13`) as `CMAKE_OSX_SYSROOT`/`-isysroot` didn't work for me.

### CODE REVIEW

**Comment by nikolaykasyanov** [comment]
This is great, I'll look into it during the upcoming weekend 👍 
Something should be done about boost build verbosity though, it seems Travis can't handle it.

**Comment by nikolaykasyanov** [comment]
@dgladkov I guess I know why boost build fails. Image used by Travis must be changed to Xcode 9.* in order to have 10.13 SDK available.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -10,10 +10,13 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
...
 
 set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9")
-set(CMAKE_OSX_SYSROOT "macosx10.12")
+set(CMAKE_OSX_SYSROOT "macosx10.13")
 set(CMAKE_CXX_STANDARD 11)
 
-execute_process(COMMAND xcrun --show-sdk-path --sdk ${CMAKE_OSX_SYSROOT} OUTPUT_VARIABLE FULL_SYSROOT_PATH OUTPUT_STRIP_TRAILING_WHITESPACE)
+# use CMAKE_OSX_SYSROOT as is if it points to existing directory, lookup full path otherwise
+if(NOT EXISTS CMAKE_OSX_SYSROOT)
+  execute_process(COMMAND xcrun --show-sdk-path --sdk ${CMAKE_OSX_SYSROOT} OUTPUT_VARIABLE CMAKE_OSX_SYSROOT OUTPUT_STRIP_TRAILING_WHITESPACE)
```
Comment (nikolaykasyanov) [inline src/CMakeLists.txt]: It should be possible to fail the whole thing if `xcrun` fails. I suppose `RESULT_VARIABLE` must be passed and checked thereafter.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -10,10 +10,13 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 endif ()
 
 set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9")
-set(CMAKE_OSX_SYSROOT "macosx10.12")
+set(CMAKE_OSX_SYSROOT "macosx10.13")
 set(CMAKE_CXX_STANDARD 11)
 
-execute_process(COMMAND xcrun --show-sdk-path --sdk ${CMAKE_OSX_SYSROOT} OUTPUT_VARIABLE FULL_SYSROOT_PATH OUTPUT_STRIP_TRAILING_WHITESPACE)
+# use CMAKE_OSX_SYSROOT as is if it points to existing directory, lookup full path otherwise
+if(NOT EXISTS CMAKE_OSX_SYSROOT)
```
Comment (nikolaykasyanov) [inline src/CMakeLists.txt]: Let's please keep `if` style consistent (i.e. a space before `(`).

**Review by nikolaykasyanov — CHANGES_REQUESTED** [review]

**Comment by dgladkov** [comment]
Lowering number of make jobs to 2x [number of cores](https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System) seems fixed OSG build timeout, but the build time difference between different runs seems to be too big to make any solid conclusions.

**Comment by nikolaykasyanov** [comment]
@dgladkov yep, it seems like Xcode 9 builds are slower (is High Sierra to blame?). Thanks for the fixes!

**Review by nikolaykasyanov — APPROVED** [review]

**Pull request merged**